u19_pipeline/recording.py

```diff
@@ -1,204 +1,253 @@
+         -> master
+         ---
+         -> acquisition.Session
+         """
+    
+     class RecordingSession(dj.Part):
+         definition = """
+         -> master
+         ---
+         -> subject.Subject
+         recording_datetime: datetime
+         """
+
+# Declare upstream ephys table ---------------------------------------------------------
+@schema
+class EphysRecordingSession(dj.Computed):
     definition = """
-    # Parameter set to be used in the preprocessing steps
-    preprocess_paramset_idx:  int(11) AUTO_INCREMENT
-    ---
-    -> RecordingModality
-    preprocess_paramset_desc: varchar(128)
-    preprocess_paramset_hash: uuid
-    unique index (preprocess_paramset_hash)
-    preprocess_paramset: longblob  # dictionary of all applicable parameters
+    -> Recording
     """
 
-    @classmethod
-    def insert_new_params(cls, recording_modality: str, preprocess_paramset_idx: int,
-                          preprocess_paramset_desc: str, preprocess_paramset: dict):
-        paramset_dict = {'recording_modality': recording_modality,
-                      'preprocess_paramset_idx': preprocess_paramset_idx,
-                      'preprocess_paramset_desc': preprocess_paramset_desc,
-                      'preprocess_paramset': preprocess_paramset,
-                      'preprocess_paramset_hash':  dict_to_uuid(preprocess_paramset)}
-        paramset_query = cls & {'preprocess_paramset_hash': paramset_dict['preprocess_paramset_hash']}
-        if paramset_query:  # If the specified param-set already exists
-            existing_paramset_idx = paramset_query.fetch1('preprocess_paramset_idx')
-            if existing_paramset_idx == preprocess_paramset_idx:  # If the existing set has the same preprocess_paramset_idx: job done
-                return
-            else:  # If not same name: human error, trying to add the same paramset with different name
-                raise dj.DataJointError(
-                    'The specified param-set'
-                    ' already exists - preprocess_paramset_idx: {}'.format(existing_paramset_idx))
-        else:
-            cls.insert1(paramset_dict)
-
-
-    def get_preprocess_params(self, preprocess_param_idx):
-        '''
-        Get process params for current recording process
-        Return:
-            preprocess_paramset (dict): preprocess params associated with recording process
-        '''
+    @property
+    def key_source(self):
+        return Recording & {'recording_modality': 'electrophysiology'}
+
+    def make(self, key):
+        self.insert1(key)
 
-        preprocess_paramset =  (self & preprocess_param_idx).fetch1('preprocess_paramset')
 
-        #If stored in MATLAB this is a numpy array to be converted to dictionary
-        if isinstance(preprocess_paramset, np.ndarray):
-            preprocess_paramset = numpy_array_to_dict(preprocess_paramset)
+# Declare upstream imaging tables ------------------------------------------------------
```

Comment:
All the imaging tables to their own imaging "pre_element" schema

Reply:
Added these tables to the `imaging_recording` module/schema.